### PR TITLE
add a java sample code for create workflow with await result

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Add a java client example for create workflow with await result.

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: samples/src/main/java/io/zeebe/example/workflow/WorkflowInstanceWithResultCreator.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+      openJobWorker(client); // open job workers so that task are executed and workflow is completed
+      System.out.println("Creating workflow instance");
+
+      final WorkflowInstanceResult workflowInstanceResult =
+          client
+              .newCreateInstanceCommand()
+              .bpmnProcessId(bpmnProcessId)
+              .latestVersion()
+              .withResult()
```
Comment (Zelldon) [inline samples/src/main/java/io/zeebe/example/workflow/WorkflowInstanceWithResultCreator.java]: Should we mark this line somehow ? with a comment that this is the interesting part

**Review by ChrisKujawa — COMMENTED** [review]
Looks good just some comments

File: docs/src/java-client-examples/workflow-instance-create-with-result.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Create a Workflow Instance and Await Result
+
+## Prerequisites
+
+1. Running Zeebe broker with endpoint `localhost:26500` (default)
+1. Run the [Deploy a Workflow example](/java-client-examples/workflow-deploy.html). Deploy `demoProcessSingleTask.bpmn` instead of `demoProcess.bpmn`
+
+## WorkflowInstanceWithResultCreator.java
+
+[Source on github](https://github.com/zeebe-io/zeebe/tree/{{commit}}/samples/src/main/java/io/zeebe/example/workflow/WorkflowInstanceWithResultCreator.java)
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply docs/src/java-client-examples/workflow-instance-create-with-result.md]: :+1:

File: samples/src/main/java/io/zeebe/example/workflow/WorkflowInstanceWithResultCreator.java (hunk 1 above)
Comment (deepthidevaki) [reply samples/src/main/java/io/zeebe/example/workflow/WorkflowInstanceWithResultCreator.java]: Yes.

**Review by ChrisKujawa — APPROVED** [review]
thanks

**Comment by deepthidevaki** [comment]
bors retry

**Pull request merged**